x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -305,60 +330,75 @@ export const ComplianceDashboard = () => {
     enabled: isCloudSecurityPostureInstalled,
   });
 
-  useEffect(() => {
-    if (hasUserSelectedTab) {
-      return;
+  const location = useLocation();
+  const history = useHistory();
+  const { services } = useKibana();
+
+  const currentTabUrlState: PosturePolicyTemplate | undefined = useMemo(() => {
+    let tab: PosturePolicyTemplate | undefined;
+    if (location.pathname === cloudPosturePages.kspm_dashboard.path) {
+      tab = POSTURE_TYPE_KSPM;
```

Comment:
you can also add the postureType as an optional parameter to the default url, and then check the value with `useParams` of react-router

[example](https://github.com/elastic/kibana/blob/897f75c789b6536ef57cabafa64f477165d5123e/x-pack/plugins/cloud_security_posture/public/common/navigation/constants.ts#L79)